pkg/domain/service.go

```diff
@@ -0,0 +1,368 @@
+	return s.Origin
+}
+
+func (s *ServiceCommon) SetOrigin(origin string) {
+	s.Origin = origin
+}
+
+func (s *ServiceCommon) IsTls() bool {
+	return s.Tls
+}
+
+func (s *ServiceCommon) SetTls(tls bool) {
+	s.Tls = tls
+}
+
+func (s *ServiceCommon) GetTlsCredentials() string {
+	return s.TlsCredentials
+}
+
+func (s *ServiceCommon) SetTlsCredentials(credential string) {
+	s.TlsCredentials = credential
+}
+
+func (s *ServiceCommon) GetIngress() AddressIngress {
+	return s.Ingress
+}
+
+func (s *ServiceCommon) SetIngress(ingress AddressIngress) {
+	s.Ingress = ingress
+}
+
+func (s *ServiceCommon) GetEgressResolvers() []EgressResolver {
+	return s.EgressResolvers
+}
+
+func (s *ServiceCommon) SetEgressResolvers(resolvers []EgressResolver) {
+	s.EgressResolvers = resolvers
+}
+
+func (s *ServiceCommon) AddEgressResolver(resolver EgressResolver) {
+	s.EgressResolvers = append(s.EgressResolvers, resolver)
+}
+
+func ValidateService(service Service) error {
+	errs := validation.IsDNS1035Label(service.GetAddress())
+	if len(errs) > 0 {
+		return fmt.Errorf("Invalid service name: %q", errs)
+	}
+
+	for _, resolver := range service.GetEgressResolvers() {
+		targets, err := resolver.Resolve()
+		if err != nil {
+			return fmt.Errorf("error resolving egresses - %w", err)
+		}
+		for _, target := range targets {
+			for _, targetPort := range target.GetPorts() {
+				if targetPort < 0 || 65535 < targetPort {
+					return fmt.Errorf("Bad target port number. Target: %s  Port: %d", resolver.String(), targetPort)
+				}
+			}
+		}
+	}
+
+	for _, port := range service.GetPorts() {
+		if port < 0 || 65535 < port {
+			return fmt.Errorf("Port %d is outside valid range.", port)
+		}
+	}
+	if service.GetAddress() != "" && service.IsEventChannel() {
+		return fmt.Errorf("Only one of aggregate and event-channel can be specified for a given service.")
+	} else if service.GetAggregate() != "" && service.GetAggregate() != "json" && service.GetAggregate() != "multipart" {
+		return fmt.Errorf("%s is not a valid aggregation strategy. Choose 'json' or 'multipart'.", service.GetAggregate())
+	} else if service.GetProtocol() != "" && service.GetProtocol() != "tcp" && service.GetProtocol() != "http" && service.GetProtocol() != "http2" {
+		return fmt.Errorf("%s is not a valid mapping. Choose 'tcp', 'http' or 'http2'.", service.GetProtocol())
+	} else if service.GetAggregate() != "" && service.GetProtocol() != "http" {
+		return fmt.Errorf("The aggregate option is currently only valid for http")
+	} else if service.IsEventChannel() && service.GetProtocol() != "http" {
+		return fmt.Errorf("The event-channel option is currently only valid for http")
+	} else if service.IsTls() && service.GetProtocol() != "http2" {
+		return fmt.Errorf("The TLS support is only available for http2")
+	} else {
+		return nil
+	}
+}
+
+func CreateRouterServiceConfig(site Site, parentRouterConfig *qdr.RouterConfig, service Service) (*qdr.RouterConfig, string, error) {
```

Comment:
Should not be. That is why I have it under `pkg/domain` and not `pkg/domain/podman`.
When we have a future `docker` implementation, or something else, this could be reused
if we need a separate router config for a given service proxy.